Mathlib/NumberTheory/NumberField/CanonicalEmbedding/Basic.lean

```diff
@@ -401,15 +402,14 @@ end norm
 
 noncomputable section stdBasis
 
-open scoped Classical
-
 open Complex MeasureTheory MeasureTheory.Measure Zspan Matrix ComplexConjugate
 
 variable [NumberField K]
 
 /-- The type indexing the basis `stdBasis`. -/
 abbrev index := {w : InfinitePlace K // IsReal w} ⊕ ({w : InfinitePlace K // IsComplex w}) × (Fin 2)
 
+open Classical in
```

Comment:
Instead of adding `open Classical in` whenever you want to have fintype instances and normed space structure on `E K`, I'd add somewhere the lines
```
instance : DecidablePred (fun (w : InfinitePlace K) ↦ IsReal w) := Classical.decPred _
instance : DecidablePred (fun (w : InfinitePlace K) ↦ IsComplex w) := Classical.decPred _
```
(maybe just after `NumberField.InfinitePlace.fintype`?)

In this way, you can remove most `open Classical` from this file and the other one.